haystack/preview/components/embedders/openai_text_embedder.py

```diff
@@ -0,0 +1,90 @@
+from typing import List, Optional, Dict, Any
+import os
+
+import openai
+
+from haystack.preview import component, default_to_dict, default_from_dict
+
+
+@component
+class OpenAITextEmbedder:
+    """
+    A component for embedding strings using OpenAI models.
+    """
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        model_name: str = "text-embedding-ada-002",
+        organization: Optional[str] = None,
+        prefix: str = "",
+        suffix: str = "",
+    ):
+        """
+        Create a OpenAITextEmbedder component.
+
+        :param api_key: The OpenAI API key. It can be explicitly provided or automatically read from the
+                        environment variable OPENAI_API_KEY (recommended).
+        :param model_name: The name of the model to use.
+        :param organization: The OpenAI-Organization ID, defaults to `None`. For more details, see OpenAI
+        [documentation](https://platform.openai.com/docs/api-reference/requesting-organization).
+        :param prefix: A string to add to the beginning of each text.
+        :param suffix: A string to add to the end of each text.
+        """
+
+        if api_key is None:
+            try:
+                api_key = os.environ["OPENAI_API_KEY"]
+            except KeyError as e:
+                raise ValueError(
+                    "OpenAITextEmbedder expects an OpenAI API key. "
+                    "Please provide one setting the environment variable OPENAI_API_KEY (recommended) or passing it explicitly."
```

Comment:
@dfokina can you have a look at this error message?

Reply:
```suggestion
                    "OpenAITextEmbedder expects an OpenAI API key. "
                    "You can set the OPENAI_API_KEY environment variable (recommended) or pass it explicitly."
```

Let's make it explicit like this.